src/main/java/org/embulk/util/config/Compat.java

```diff
@@ -113,21 +127,50 @@ static ObjectNode rebuildObjectNode(final DataSource source) throws IOException
             }
             throw new IllegalStateException("DataSource(Impl)#toJson() threw unexpected Exception.", targetException);
         } catch (final IllegalAccessException ex) {
-            logger.debug("DataSource(Impl)#toJson is not accessible unexpectedly. DataSource: {}, toJson: {}, ",
-                         source.getClass(), toJson);
-            throw new IllegalStateException("DataSource(Impl)#toJson() is not accessible.", ex);
+            // "org.embulk.util.config.DataSourceImpl" fo embulk-util-config 0.1.4 or earlier was package-private,
+            // then invoking "toJson" of "org.embulk.util.config.DataSourceImpl" caused IllegalAccessException.
+            // https://github.com/embulk/embulk-util-config/pull/20
+            //
+            // Pass-through to the next step to setAccessible(true) for the "toJson" Method instance.
```

Comment:
Q: I assumed a pass-through here means there's a possibility of another type of exception is thrown but could be resolved with the "accessible = true" hack?

Reply:
My thought is,

If `toJson.invoke(source);` throws a type of Exception:
1. If the Exception is `InvocationTargetException`: all rethrown to fail explicittly.
2. If the Exception is `IllegalAccessException`: passing-through intentionally.
3. Otherwise: thrown-up to fail.

And those are intended and expected. The case 1 and case 3 should fail immediately. Only the case 2 should pass-through to the next step.

Then, only in the case 2, the accessible flag should be set back `false` by `toJson.setAccessible(false);` in `finally`.